5.0.0 fails to build on FreeBSD i386 due to s32/time_t type aliasing

##### Issue type
 - Build issue
 
##### Minetest version
```
5.0.0
```

##### OS / Hardware
Operating system: FreeBSD 12.0-RELEASE i386

##### Summary
```
In file included from /wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/content_sao.cpp:28:
In file included from /wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/server.h:37:
In file included from /wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/clientiface.h:26:
/wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/network/networkpacket.h:106:17: error: class member cannot be redeclared
        NetworkPacket &operator>>(s32 &dst);
                       ^
/wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/network/networkpacket.h:91:17: note: previous declaration is here
        NetworkPacket &operator>>(std::time_t &dst);
                       ^
/wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/network/networkpacket.h:107:17: error: class member cannot be redeclared
        NetworkPacket &operator<<(s32 src);
                       ^
/wrkdirs/usr/ports/games/minetest/work/minetest-5.0.0/src/network/networkpacket.h:92:17: note: previous declaration is here
        NetworkPacket &operator<<(std::time_t src);
                       ^
2 errors generated.
```

Because `time_t` type is unspecified, it may be any integral integer type capable of holding number of seconds since Jan 1 1970. On FreeBSD i386, it's the same type as signed 32 bit integer.

##### Steps to reproduce
Always reproducible on FreeBSD i386.

--- a/src/network/clientpackethandler.cpp
+++ b/src/network/clientpackethandler.cpp
@@ -408,7 +408,9 @@ void Client::handleCommand_ChatMessage(NetworkPacket *pkt)
 		return;
 	}
 
-	*pkt >> chatMessage->sender >> chatMessage->message >> chatMessage->timestamp;
+	u64 timestamp;
+	*pkt >> chatMessage->sender >> chatMessage->message >> timestamp;
+	chatMessage->timestamp = static_cast<std::time_t>(timestamp);
 
 	chatMessage->type = (ChatMessageType) message_type;
 

--- a/src/network/networkpacket.cpp
+++ b/src/network/networkpacket.cpp
@@ -281,12 +281,6 @@ NetworkPacket& NetworkPacket::operator<<(u64 src)
 	return *this;
 }
 
-NetworkPacket& NetworkPacket::operator<<(std::time_t src)
-{
-	*this << (u64) src;
-	return *this;
-}
-
 NetworkPacket& NetworkPacket::operator<<(float src)
 {
 	checkDataSize(4);
@@ -363,16 +357,6 @@ NetworkPacket& NetworkPacket::operator>>(u32& dst)
 }
 
 NetworkPacket& NetworkPacket::operator>>(u64& dst)
-{
-	checkReadOffset(m_read_offset, 8);
-
-	dst = readU64(&m_data[m_read_offset]);
-
-	m_read_offset += 8;
-	return *this;
-}
-
-NetworkPacket& NetworkPacket::operator>>(std::time_t& dst)
 {
 	checkReadOffset(m_read_offset, 8);
 

--- a/src/network/networkpacket.h
+++ b/src/network/networkpacket.h
@@ -19,7 +19,6 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 
 #pragma once
 
-#include <ctime>
 #include "util/pointer.h"
 #include "util/numeric.h"
 #include "networkprotocol.h"
@@ -87,9 +86,6 @@ public:
 
 	NetworkPacket &operator>>(u64 &dst);
 	NetworkPacket &operator<<(u64 src);
-
-	NetworkPacket &operator>>(std::time_t &dst);
-	NetworkPacket &operator<<(std::time_t src);
 
 	NetworkPacket &operator>>(float &dst);
 	NetworkPacket &operator<<(float src);

--- a/src/server.cpp
+++ b/src/server.cpp
@@ -1559,7 +1559,7 @@ void Server::SendChatMessage(session_t peer_id, const ChatMessage &message)
 	NetworkPacket pkt(TOCLIENT_CHAT_MESSAGE, 0, peer_id);
 	u8 version = 1;
 	u8 type = message.type;
-	pkt << version << type << std::wstring(L"") << message.message << message.timestamp;
+	pkt << version << type << std::wstring(L"") << message.message << (u64)message.timestamp;
 
 	if (peer_id != PEER_ID_INEXISTENT) {
 		RemotePlayer *player = m_env->getPlayer(peer_id);

